shims/spark311/src/main/scala/com/nvidia/spark/rapids/shims/spark311/Spark311Shims.scala

```diff
@@ -233,7 +233,7 @@ class Spark311Shims extends Spark301Shims {
         }),
       GpuOverrides.exec[InMemoryTableScanExec](
         "Implementation of InMemoryTableScanExec to use GPU accelerated Caching",
-        ExecChecks(TypeSig.commonCudfTypes, TypeSig.all),
+        ExecChecks(TypeSig.commonCudfTypes + TypeSig.DECIMAL + TypeSig.STRUCT, TypeSig.all),
```

Comment:
Doesn't this need a `.nested()` to indicate it supports all of these types as nested types as well?

Reply:
Yes, good catch jason. I missed it because most of the execs already have nested appended to the supported types.